packages/fs-detectors/src/detectors/filesystem.ts

```diff
@@ -125,4 +125,15 @@ export abstract class DetectorFilesystem {
   public chdir = (name: string): DetectorFilesystem => {
     return this._chdir(name);
   };
+
+  /**
+   * Writes a file to the filesystem cache.
+   * @param name the name of the file to write
+   * @param content The content of the file
+   */
+  public writeFile(name: string, content: string): void {
+    this.readFileCache.set(name, Promise.resolve(Buffer.from(content)));
+    this.fileCache.set(name, Promise.resolve(true));
+    this.pathCache.set(name, Promise.resolve(true));
```

Comment:
Do we need to invoke `this._writefile()` too so the class inheriting this abstract class can change the behavior?

Reply:
I'd say we should revisit this when there's a use-case for it.  Was trying to avoid a major version bump

🤔 I could see the function signature changing to `public async writeFile(...): Promise<void>` if we wanted to support `this._writeFile()`. 

I'll go ahead and update the function signature for now to allow us in the future to add support for `this._writeFile()`